`make all` fails in `hack/update_codegen.sh`

**What happened**:

I ran `make all` as [instructed](https://github.com/kudobuilder/kudo/blob/master/CONTRIBUTING.md#build-instructions).
It failed after a few minutes:
```
./hack/update_codegen.sh
go: finding github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78
[...]
go: finding sourcegraph.com/sqs/pbtypes v0.0.0-20180604144634-d3ebe8f20ae4
Cloning into 'hack/code-gen/8e001e5d1894'...
remote: Enumerating objects: 279, done.
remote: Counting objects: 100% (279/279), done.
remote: Compressing objects: 100% (187/187), done.
remote: Total 7519 (delta 141), reused 199 (delta 85), pack-reused 7240
Receiving objects: 100% (7519/7519), 4.43 MiB | 391.00 KiB/s, done.
Resolving deltas: 100% (3670/3670), done.
HEAD is now at 8e001e5 Merge pull request #83433 from liggitt/automated-cherry-pick-of-#83261-upstream-release-1.16
go mod init: go.mod already exists
./hack/update_codegen.sh: linia 22: hack/code-gen/8e001e5d1894/generate-groups.sh: Nie ma takiego pliku ani katalogu
Makefile:84: polecenia dla obiektu 'generate' nie powiodły się
make: *** [generate] Błąd 127
```

**What you expected to happen**:

Success.

**How to reproduce it (as minimally and precisely as possible)**:

Follow the documented [build instructions](https://github.com/kudobuilder/kudo/blob/master/CONTRIBUTING.md#build-instructions).

After fixing that, the next failure is:
```
./hack/update_codegen.sh
Using cached code generator version: 8e001e5d1894
Generating deepcopy funcs
hack/code-gen/8e001e5d1894/generate-groups.sh: line 71: GOPATH: unbound variable
Makefile:84: recipe for target 'generate' failed
make: *** [generate] Error 1
```

--- a/hack/update_codegen.sh
+++ b/hack/update_codegen.sh
@@ -15,7 +15,12 @@ if [ -d "$CODE_GEN_DIR" ]; then
     echo "Using cached code generator version: $VERSION"
 else
     git clone https://github.com/kubernetes/code-generator.git "${CODE_GEN_DIR}"
-    cd "${CODE_GEN_DIR}" && git reset --hard "${VERSION}" && go mod init && cd -
+    cd "${CODE_GEN_DIR}"
+    git reset --hard "${VERSION}"
+    if [[ ! -f go.mod ]]; then
+        go mod init
+    fi
+    cd -
 fi
 
 "${CODE_GEN_DIR}"/generate-groups.sh \

